docs/machines-as-data.md

```
@@ -327,13 +327,13 @@ A run settles in one of two ways.
 result = await runAgent(machine, { snapshot, event: { type: "APPROVE" }, executors });
 ```
 
-> **Note:** An **idle state** is any state with no `invoke`. Nothing runs in it, so the machine waits for an external event declared under `on`. A state with an `invoke` is doing work: a decision, a text request, or an `agent.userInput` pause.
+> **Note:** By default, an **idle state** is an active snapshot that accepts an external event anywhere in its active hierarchy, or has `meta.interaction`. `runAgent` also verifies that no invoked child, eventless transition, or delayed transition is still working.
 
```

markdownlint-cli2 (0.23.2)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)